src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/readers/JsonReader.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.sling.feature.cpconverter.handlers.slinginitialcontent.readers;
+
+import org.apache.jackrabbit.commons.SimpleValueFactory;
+import org.apache.jackrabbit.oak.spi.security.authorization.restriction.RestrictionDefinition;
+import org.apache.sling.feature.cpconverter.accesscontrol.RestrictionProvider;
+import org.apache.sling.jcr.contentloader.ContentCreator;
+
+import javax.jcr.*;
+import javax.json.*;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class  JsonReader extends org.apache.sling.jcr.contentloader.internal.readers.JsonReader {
+    
+    private static final String SECURITY_PRINCIPLES = "security:principals";
+    private static final String SECURITY_ACL = "security:acl";
+ 
+    private ValueFactory factory = new SimpleValueFactory();
```

Comment:
what exactly do you need the valuefactory for? i would avoid that and just read the string(?) value plus type information from the initial-content.
repoinit just reads all restrictions as string values and then will convert it to the required type as needed (in the jcr-impl where it has a session and a value factory at hand)